src/js/player.js

```diff
@@ -2679,57 +2679,138 @@ class Player extends Component {
+   * by an external function).
+   *
+   * @param {EventTarget~Event} event
+   *        The `keydown` event that caused this function to be called.
+   *
+   * @listens keydown
    */
   handleKeyPress(event) {
 
-    if (this.options_.hotkeys !== false) {
+    if (this.options_.userActions && this.options_.userActions.hotkeys && (this.options_.userActions.hotkeys !== false)) {
 
-      if (typeof this.options_.hotkeys === 'function') {
+      if (typeof this.options_.userActions.hotkeys === 'function') {
 
-        this.options_.hotkeys(event);
+        this.options_.userActions.hotkeys.call(this, event);
 
-      // Trigger the ControlBar equivalent for each key, if possible
-      } else if (keycode.isEventKey(event, 'f')) {
-        // "f" = toggle fullscreen
+      } else {
 
-        event.preventDefault();
+        this.handleHotkeys(event);
 
-        const cb = this.player_.controlBar;
-        const fullscreenToggle = cb && cb.fullscreenToggle;
+      }
+    }
+  }
 
-        if (fullscreenToggle && fullscreenToggle.handleClick) {
-          fullscreenToggle.handleClick();
-        }
-      } else if (keycode.isEventKey(event, 'm')) {
-        // "m" = toggle mute
+  /**
+   * Called when this Player receives a hotkey keydown event.
+   * Supported player-wide hotkeys are:
+   *
+   *   f          - toggle fullscreen
+   *   m          - toggle mute
+   *   k or Space - toggle play/pause
+   *
+   * @param {EventTarget~Event} event
+   *        The `keydown` event that caused this function to be called.
+   */
+  handleHotkeys(event) {
+    const hotkeys = this.options_.userActions ? this.options_.userActions.hotkeys : null;
+    let fullscreenKey;
+    let muteKey;
+    let playPauseKey;
 
-        event.preventDefault();
+    if (hotkeys) {
+      fullscreenKey = hotkeys.fullscreenKey;
+      muteKey = hotkeys.muteKey;
+      playPauseKey = hotkeys.playPauseKey;
+    }
 
-        const cb = this.player_.controlBar;
-        const volumePanel = cb && cb.volumePanel;
-        const muteToggle = (cb && cb.muteToggle) || (volumePanel && volumePanel.muteToggle);
+    if (typeof fullscreenKey !== 'function') {
+      fullscreenKey = keydownEvent => keycode.isEventKey(keydownEvent, 'f');
+    }
+    if (typeof muteKey !== 'function') {
+      muteKey = keydownEvent => keycode.isEventKey(keydownEvent, 'm');
+    }
+    if (typeof playPauseKey !== 'function') {
+      playPauseKey = keydownEvent => (keycode.isEventKey(keydownEvent, 'k') || keycode.isEventKey(keydownEvent, 'Space'));
+    }
 
-        if (muteToggle && muteToggle.handleClick) {
-          muteToggle.handleClick();
-        } else {
-          this.player_.muted(this.player_.muted() ? false : true);
-        }
-      } else if (keycode.isEventKey(event, 'k')) {
-        // "k" = toggle play/pause
+    // Trigger the ControlBar equivalent for each key, if possible
+    const cb = this.controlBar;
 
-        event.preventDefault();
+    if (fullscreenKey.call(this, event)) {
```

Comment:
if someone overrides the fullscreen key with a method that returns false, this will try to evaluate the other methods, though, maybe not a big deal.